pycoal/mineral.py

```diff
@@ -91,44 +92,64 @@ def classify_image(self, image_file_name, classified_file_name):
-                    resampled_pixel = numpy.nan_to_num(resample(pixel))
-
-                    # calculate spectral angles
-                    angles = spectral.spectral_angles(resampled_pixel[numpy.newaxis,
-                                                                     numpy.newaxis,
-                                                                     ...],
-                                                      self.library.spectra)
-
-                    # normalize confidence values from [pi,0] to [0,1]
-                    for z in range(angles.shape[2]):
-                        angles[0,0,z] = 1-angles[0,0,z]/math.pi
-
-                    # get index of class with largest confidence value
-                    index_of_max = numpy.argmax(angles)
-
-                    # classify pixel if confidence above threshold
-                    if angles[0,0,index_of_max] > self.threshold:
-
-                        # index from one (after zero for no data)
-                        classified[x,y] = index_of_max + 1
-
+            classified = numpy.zeros(shape=(M,N), dtype=numpy.uint16)
+
+            # for each pixel in the image
+            for x in range(M):
+
+                for y in range(N):
+
+                    # read the pixel from the file
+                    pixel = data[x,y]
+
+                    # if it is not a no data pixel
+                    if not numpy.isclose(pixel[0], -0.005) and not pixel[0]==-50:
+
+                        # resample the pixel ignoring NaNs from target bands that don't overlap
+                        # TODO fix spectral library so that bands are in order
+                        resampled_pixel = numpy.nan_to_num(resample(pixel))
+
+                        # calculate spectral angles
+                        angles = spectral.spectral_angles(resampled_pixel[numpy.newaxis,
+                                                                         numpy.newaxis,
+                                                                         ...],
+                                                          self.library.spectra)
+
+                        # normalize confidence values from [pi,0] to [0,1]
+                        for z in range(angles.shape[2]):
+                            angles[0,0,z] = 1-angles[0,0,z]/math.pi
+
+                        # get index of class with largest confidence value
+                        index_of_max = numpy.argmax(angles)
+
+                        # classify pixel if confidence above threshold
+                        if angles[0,0,index_of_max] > self.threshold:
+
+                            # index from one (after zero for no data)
+                            classified[x,y] = index_of_max + 1
+
+        elif classifier_type == 'GML':
+            # Gaussian Maximum Likelihood
+            # the following is the method layed out in the spectral python documentation, needs to be converted to work with our formatting
```

Comment:
Don't just comment that it needs to be converted... actually go ahead and convert it.